core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategy.java

```diff
@@ -88,15 +102,35 @@ static RetryStrategy onException(Function<? super Throwable, ? extends Backoff>
     /**
      * Returns a {@link RetryStrategy} that retries with the {@link Backoff#ofDefault()}
      * when the response status is 5xx (server error) or an {@link Exception} is raised.
+     *
+     * @deprecated Use {@link RetryRuleBuilder#onServerErrorStatus()}}, {@link RetryRuleBuilder#onException()}
+     *             and {@link RetryRuleBuilder#thenBackoff()}.
+     *             For example:
+     *             <pre>{@code
+     *             RetryRule.onServerErrorStatus()
+     *                      .onException()
+     *                      .thenBackoff();
+     *             }</pre>
      */
+    @Deprecated
     static RetryStrategy onServerErrorStatus() {
```

Comment:
I think most of the builder factories aren't so useful, but having an equivalent factory in the new class that returns a very common `RetryRule` will be great.